src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/prometheus/silence-list/silence-list.component.ts

```diff
@@ -156,6 +173,26 @@ export class SilenceListComponent extends PrometheusListHelper {
     this.selection = selection;
   }
 
+  getAlerts(silences: any) {
+    if (!this.visited) {
+      this.visited = true;
+      const rules = this.silenceFormComponent.getRules();
+      silences.forEach((silence: any) => {
+        if (silence.status.state !== 'expired') {
+          silence.silencedAlerts = [];
+          silence.matchers.forEach((matcher: any) => {
+            this.rules = this.silenceMatcher.getMatchedRules(matcher, rules);
+            const alertNames: string[] = [];
+            for (let i = 0; i < this.rules.length; i++) {
+              alertNames.push(this.rules[i].name);
+            }
+            silence.silencedAlerts = alertNames;
+          });
+        }
+      });
+    }
```

Comment:
this is a complex function with a lot of for loop inside for loops. can we improve this?

Reply:
Sure